src/main/src/main/java/org/geoserver/security/SecuredResourceNameChangeListener.java

```diff
@@ -91,61 +89,75 @@ public void handleModifyEvent(CatalogModifyEvent event) throws CatalogException
+        if (eventSource instanceof WorkspaceInfo) {
+            // if workspace has been renamed
+            filter = filter(oldName::equalsIgnoreCase, l -> true);
+            updater = r -> r.setRoot(newName);
+        } else if (eventSource instanceof ResourceInfo) {
             // if a layer has been renamed
             // similar layer names can exist in different workspaces
-            resourceInfo = event.getSource();
-            resourceWorkSpaceName =
-                    ((ResourceInfo) resourceInfo).getStore().getWorkspace().getName();
-
-        } else if (event.getSource() instanceof LayerGroupInfo) {
+            String wsName = ((ResourceInfo) eventSource).getStore().getWorkspace().getName();
+            filter = filter(wsName::equalsIgnoreCase, oldName::equalsIgnoreCase);
+            updater = r -> r.setLayer(newName);
+        } else if (eventSource instanceof LayerGroupInfo) {
             // if a layergroup has been renamed
             // similar layergroup names can exist in different workspaces
-            resourceInfo = event.getSource();
-            resourceWorkSpaceName = ((LayerGroupInfo) resourceInfo).getWorkspace().getName();
+            LayerGroupInfo lg = (LayerGroupInfo) eventSource;
+            WorkspaceInfo ws = lg.getWorkspace();
+            if (ws == null) {
+                filter = filter(oldName::equalsIgnoreCase, r -> true);
+                updater = r -> r.setRoot(newName);
+            } else {
+                filter = filter(ws.getName()::equalsIgnoreCase, oldName::equalsIgnoreCase);
+                updater = r -> r.setLayer(newName);
+            }
+        } else {
+            return;
         }
 
-        if (resourceWorkSpaceName != null && resourceInfo != null) {
-            // if a security rule exists for the layer whose name has been changed
-            if (layerHasSecurityRule(resourceWorkSpaceName, oldName)) {
-                LOGGER.info(
+        apply(
+                filter,
+                updater,
+                () ->
                         String.format(
-                                "Updating Security Rules for Renamed Feature Type: %s", oldName));
+                                "Updating Security Rules for Renamed %s: %s",
+                                eventSource instanceof WorkspaceInfo ? "Workspace" : "Feature Type",
+                                oldName));
+    }
 
-                // get rules in which workspace name should be updated
-                List<DataAccessRule> rulesToModifyList = getDataAccessRule(resourceInfo, oldName);
-                for (DataAccessRule rule : rulesToModifyList) rule.setLayer(newName);
-            }
+    void apply(
+            Predicate<DataAccessRule> filter,
+            Consumer<DataAccessRule> action,
+            Supplier<String> logMessage) {
+        List<DataAccessRule> matches = getDataAccessRules(filter);
+        if (!matches.isEmpty()) {
+            LOGGER.info(logMessage.get());
```

Comment:
Better give it the supplier directly, it can skip the message production if the LOGGER level is higher.

Reply:
good catch, done.